test/acceptance/graphql_resolvers/local_aggregate_groupby_nearobject_test.go

```diff
@@ -0,0 +1,166 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/semi-technologies/weaviate/test/helper"
+	graphqlhelper "github.com/semi-technologies/weaviate/test/helper/graphql"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func aggregatesArrayClassWithGroupingAndNearObject(t *testing.T) {
+	tests := []struct {
+		name     string
+		property string
+		expected func(valueString string, count interface{})
+	}{
+		{
+			name:     "ints",
+			property: "ints",
+			expected: func(valueString string, count interface{}) {
+				if valueString == "1" {
+					assert.Equal(t, json.Number("3"), count)
+				}
+				if valueString == "2" {
+					assert.Equal(t, json.Number("2"), count)
+				}
+				if valueString == "3" {
+					assert.Equal(t, json.Number("1"), count)
+				}
```

Comment:
Feel free to ignore, but I think this would make the tests more readable:

could we have a function 
func expectedResults(valueString string, count interface{}, expected_number string) {
assert.Equal(t, json.Number(expected_number), count)
}

and then do 
expected: map[string]string{"1":"3", "2":"2", "3": "1"}

and in the test
expectedResults(valuestring, count, tt.expected[valuestring])

Reply:
Good point. I will do this change